core/src/main/java/org/jenkins/ui/symbol/Symbol.java

```diff
@@ -0,0 +1,111 @@
+package org.jenkins.ui.symbol;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Functions;
+import hudson.PluginWrapper;
+import hudson.Util;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.model.Jenkins;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.jenkins.ui.icon.IconSet;
+
+/**
+ * Helper class to load symbols from Jenkins core or plugins.
+ * @since TODO
+ */
+public final class Symbol {
+    private static final Logger LOGGER = Logger.getLogger(Symbol.class.getName());
+    // keyed by plugin name / core, and then symbol name returning the SVG as a string
+    private static final Map<String, Map<String, String>> SYMBOLS = new ConcurrentHashMap<>();
+    static final String PLACEHOLDER_SVG =
+            "<svg xmlns=\"http://www.w3.org/2000/svg\" class=\"ionicon\" height=\"48\" viewBox=\"0 0 512 512\"><title>Close</title><path fill=\"none\" stroke=\"currentColor\" stroke-linecap=\"round\" stroke-linejoin=\"round\" stroke-width=\"32\" d=\"M368 368L144 144M368 144L144 368\"/></svg>";
+
+    /**
+     * A substring of the placeholder so that tests can match it.
+     */
+    static final String PLACEHOLDER_MATCHER = "M368 368L144 144M368 144L144 368";
+
+    private Symbol() {}
+
+    /**
+     * Generates the svg markup for the given symbol name and attributes.
+     * @param request the symbol request object.
+     * @return The svg markup for the symbol.
+     * @since TODO
+     */
+    public static String get(@NonNull SymbolRequest request) {
+        String name = request.getName();
+        String title = request.getTitle();
+        String tooltip = request.getTooltip();
+        String classes = request.getClasses();
+        String pluginName = request.getPluginName();
+        String id = request.getId();
+        String translatedName = IconSet.cleanName(name);
```

Comment:
Would it make more sense to move the compatibility code out of this new class and into callers that potentially pass legacy values? Then guaranteed modern callers wouldn't have this hack on the code path.